server/src/main/java/org/opensearch/index/remote/RemoteStoreEnums.java

```diff
@@ -92,14 +92,17 @@ public String getName() {
     public enum PathType {
         FIXED(0) {
             @Override
-            public BlobPath generatePath(PathInput pathInput, PathHashAlgorithm hashAlgorithm) {
+            public BlobPath generatePath(RemoteStorePathStrategy.BasePathInput pathInput, PathHashAlgorithm hashAlgorithm) {
```

Comment:
the if/else logic across the class does not look right.

Reply:
Abstracted out path creation logic to BasePathInput class, with override in PathInput. This removes the need for if/else here.

Reply:
Had kept the name for BWC, if thats not an issue then we can keep that as parent class